src/anemoi/inference/checks.py

```diff
@@ -30,6 +33,8 @@ def check_data(title: str, data: Any, variables: list[str], dates: list[datetime
         The list of variable names.
     dates : List[datetime.datetime]
         The list of dates.
+    checkpoint : Any
+        The checkpoint
```

Comment:
Is checkpoint really an `Any`, use the object

```suggestion
from anemoi.utils.humanize import plural
from anemoi.inference.checkpoint import Checkpoint
from earthkit.data.utils.dates import to_datetime

LOG = logging.getLogger(__name__)


def check_data(title: str, data: Any, variables: list[str], dates: list[datetime.datetime], checkpoint: Checkpoint) -> None:
"""Check if the data matches the expected number of fields based on variables and dates.

   Parameters
@@ -30,6 +33,8 @@ def check_data(title: str, data: Any, variables: list[str], dates: list[datetime
       The list of variable names.
   dates : List[datetime.datetime]
       The list of dates.
    checkpoint : Checkpoint
        The checkpoint
```